core/objects.go

```diff
@@ -347,6 +348,13 @@ func (ch Challenge) checkConsistency() error {
 	return nil
 }
 
+func (ch Challenge) GenerateID() string {
+	h := sha256.New()
```

Comment:
Let's actually use [hash/fnv](https://golang.org/pkg/hash/fnv/) since it's faster and we don't care about cryptographic security.